backend/question_service/controllers/questionController.js

```diff
@@ -181,10 +182,55 @@ const deleteQuestion = async (req, res) => {
   }
 };
 
+// get questions from an array of question ids
+const getQuestions = async (req, res) => {
+  try {
+    const questionIds = req.body.questionIds;
+
+    if (!questionIds) {
+      return res.status(400).json({
+        message: "Question IDs is (are) required",
+      });
+    }
+
+    const questionDataArray = []; // push each qn data into here
+
+    for (const questionId of questionIds) {
+      const questionDoc = await db
+        .collection("questions")
+        .doc(questionId)
+        .get();
+
+      const questionData = questionDoc.data();
+
+      if (!questionData) {
+        console.log("Question doesn't exists!");
+        continue;
+      }
+
+      const tempQuestion = {
+        title: questionData.title,
+        description: questionData.description,
+        difficulty: questionData.difficulty,
+        topics: questionData.topics,
+        examples: questionData.examples,
+        constraints: questionData.constraints,
+      };
```

Comment:
I think we can just return the entire question Object without removing the id field when returning the question object. Even though the receiver knows of the question ids, he does not know which questionData maps to which question ids.

Reply:
![image](https://github.com/user-attachments/assets/dcc8bf42-6184-4a23-82fe-3bd28d7ea4f3)
firebase returns question data w/o ID, updated function to include 1. questionId; and 2. dateCreated; in the object to be returned